### ISSUE TITLE
Update to collector v0.24.0

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-operator#238]
Same as #231, but for 0.24.0. Changelog: https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.24.0

**Comment by jpkrohling** [comment]
There are a few breaking changes to this one, requiring an upgrade routine.

**Comment by jnodorp-jaconi** [comment]
Taking this one.

**Comment by jnodorp-jaconi** [comment]
> There are a few breaking changes to this one, requiring an upgrade routine.

I am not too familiar with opentelemetry-collector, but the breaking changes listed in the [changelog](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.24.0) all look like breaking changes to the Go API. The operator should not be affected by those.

Am I missing something?

**Comment by jpkrohling** [comment]
This one is a good candidate to be in the upgrade routine:

https://github.com/open-telemetry/opentelemetry-collector/pull/2782

I went over the other changes and I don't see any other change that we should take into consideration.

**Comment by jpkrohling** [comment]
Oh, and thanks for volunteering!

**Comment by jnodorp-jaconi** [comment]
> This one is a good candidate to be in the upgrade routine:
> 
> [open-telemetry/opentelemetry-collector#2782](https://github.com/open-telemetry/opentelemetry-collector/pull/2782)
> 
> I went over the other changes and I don't see any other change that we should take into consideration.

Good catch! I only looked at the changes marked as breaking. I‘ll add an upgrade routine!
